ejb-txn-remote-call/README.adoc

```diff
@@ -0,0 +1,1093 @@
+cp -r ${jbossHomeName} server3
+{jbossHomeName}_3="$PWD/server3"
+----
+
+[#_configure_the_credentials_for_remote_call_authentication]
+=== Configure the credentials for remote call authentication
+
+To successfully process the remote call  from `server1` to either `server2`
+or to `server3` you must create a user on the receiver server
+to authenticate the call.
+
+Run the following procedure in the directories `__{jbossHomeName}_2__` and `__{jbossHomeName}_3__` to create the users for `server2` and `server3`.
+
+[#add_application_user]
+// Add the Authorized Application User
+include::../shared-doc/add-application-user.adoc[leveloffset=+3]
+
+[NOTE]
+====
+For the `add-user.sh` (or `.bat`) command you, can add the parameter `-ds`.
+When you include this parameter, after the user is added, the system outputs a secret value that you use when setting up the remote output connection on `server1`.
+
+The `-ds` parameter outputs:
+
+[code,sh]
+----
+To represent the user add the following to the server-identities definition <secret value="cXVpY2tzdGFydFB3ZDEh" />
+----
+====
+
+You must configure `server1`.
+See the script `${PATH_TO_QUICKSTART_DIR}/ejb-txn-remote-call/client/extensions/remote-configuration.cli`
+to review the commands that will be executed.
+The `cli` script is configured with `cli.local.properties`
+to run in https://wildfly.org/news/2015/03/13/Offline-CLI/[embedded mode] against the `standalone.xml`.
+
+[[remote_configuration_cli]]
+The CLI script `remote-configuration.cli` configures several options in the application server.
+
+* It configures a `remote outbound socket` that points to the port where EJB remoting endpoint
+  can be reached at `server2`.
+* It configures a https://wildscribe.github.io/WildFly/18.0/subsystem/remoting/remote-outbound-connection/index.html[`remote outbound connection`]
+  that is referenced in the war deployment for connecting to the remote server via EJB remoting
+  (see `${PATH_TO_QUICKSTART_DIR}/ejb-txn-remote-call/client/src/main/webapp/WEB-INF/jboss-ejb-client.xml`).
+* It defines the security realm that uses the credentials added in the section <<add_application_user, Add the Authorized Application User>>.
+* Then the deployment of `server1` is updated with the property `-Dwildfly.config.url` pointing to the configuration file `custom-config.xml`
+  that refers to application user's credentials.
+  The `custom-config.xml` descriptor provides the ability to authenticate transaction recovery after a transaction fails.
+
+[source,sh,subs="+quotes,attributes+"]
+----
+# go to the directory with distribution of server1
+cd ${jbossHomeName}_1
+./bin/jboss-cli.sh \
+  --file=${PATH_TO_QUICKSTART_DIR}/ejb-txn-remote-call/client/extensions/remote-configuration.cli \
+  --properties=${PATH_TO_QUICKSTART_DIR}/ejb-txn-remote-call/client/extensions/cli.local.properties
+----
+
+NOTE: For Windows, use the `bin\jboss-cli.bat` script.
+
+[#_configure_datasources]
+=== Configure datasources
```

Comment:
Due to the complexity and length of the document I suggest you to stick with with the default datasource on bare metal, and whatever is simpler to use on OpenShift.